lib/Platform/ElasticSearchPlatform.php

```diff
@@ -378,7 +378,9 @@ private function cleanHost(string $host): string {
 			return $host;
 		}
 
-		return trim(rtrim($host, '/'));
+		$trimmed = trim(rtrim($host, '/'));
+		// remove username and password from url if present
+		return preg_replace('/^(https?:\/\/)([^:]+:[^@]+@)/', '$1', $trimmed);
```

Comment:
Example of a valid URL that gets destroyed by this RegEx: `https://example.org:9000/ignore_th@t`. 

Imho using a custom RegEx for that purpose is brittle and makes it harder to reason about the code and the edge cases.

More broadly, what is the point of removing the username and the password from the host string? ES ignores them just fine when parsing the URL. Imho we should either find an elegant & robust solution or just not tinker with the URL.

Reply:
Hi @devnoname120 and thanks for your in-depth review. Based on your feedback, I did some tests and indeed I can confirm that the following sample code works with the used ES client v8.6.1:

```php
<?php
declare(strict_types=1);

require 'vendor/autoload.php';

use Elastic\Elasticsearch\ClientBuilder;

try {
    $client = ClientBuilder::create()
        ->setHosts(['http://elastic:mypass@elasticsearch_nc:9200'])
        ->setBasicAuthentication('elastic', 'mypass')
        ->build();
    $query = [
        'index' => 'nextcloud',
        'body' => [
            'query' => [
                'query_string' => [
                    'query' => 'ppv'
                ]
            ]
        ]
    ];
    $result = $client->search($query)->asArray();
}
catch(Exception $e){
    echo $e->getMessage();
}
```

**I think the key takeaways here are:**

* There's no need to "clean" the ES hosts
* There's also no need to delete the (eventually present) basic auth info from the ES host string
* The `cleanHost` function is not really needed anymore IMHO

I will refactor the code accordingly, thank you

Btw: bare with me, I've never used the ES php client before and I just tried to get things up and running :smile: